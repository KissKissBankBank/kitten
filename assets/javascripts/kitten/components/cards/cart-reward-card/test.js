import React from 'react'
import renderer from 'react-test-renderer'
import { CartRewardCard } from './index'

describe('<CartRewardCard />', () => {
  let component

  describe('ensure backward compatibility', () => {
    // Desactivate warnings.
    jest.spyOn(global.console, 'error').mockImplementation(() => {})

    const component = renderer
      .create(
        <CartRewardCard
          titleAmount="Custom title amount"
          textDescription="Custom text description"
        />,
      )
      .toJSON()

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('by default', () => {
    const component = renderer
      .create(
        <CartRewardCard titleAmount="Custom title amount">
          <CartRewardCard.Description>
            Custom text description
          </CartRewardCard.Description>
        </CartRewardCard>,
      )
      .toJSON()

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with some props', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <CartRewardCard
            titleAmount="Custom title amount"
            titleTag="Custom title tag"
            subtitle="Custom subtitle"
            updateAmountTitle="Custom amount"
            updateAmountLink="Custom amount link"
          >
            <CartRewardCard.Information
              title="Custom shipping title"
              value="Custom shipping value"
            />
            <CartRewardCard.Description>
              Custom text description
            </CartRewardCard.Description>
          </CartRewardCard>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
