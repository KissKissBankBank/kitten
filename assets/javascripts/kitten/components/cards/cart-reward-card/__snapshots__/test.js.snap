// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartRewardCard /> by default matches with snapshot 1`] = `
<div
  className="sc-ifAKCX bpiQjo k-CartRewardCard"
  onAnimationEnd={[Function]}
>
  <div
    className="k-CartRewardCard__cart"
  >
    <button
      aria-label="Delete"
      className="k-ButtonIcon k-CartRewardCard__garbage k-ButtonIcon--hydrogen k-ButtonIcon--micro"
      onClick={[Function]}
      tabIndex={null}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="k-ButtonIcon__svg"
        fill="#fff"
        height="14"
        viewBox="0 0 12 14"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <path
          d="M4.3 0h3v2h-3zM0 2h12v1H0zm1 2l1 10h8l1-10H1zm3.5 8h-1V6h1v6zm2 0h-1V6h1v6zm2 0h-1V6h1v6z"
        />
      </svg>
    </button>
  </div>
  <div
    className="k-u-margin-bottom-quadruple k-CartRewardCard__description"
  >
    <div
      className="k-u-margin-bottom-double k-u-margin-bottom-single@xs-down"
    >
      <h1
        className="sc-bwzfXH AMZiG k-Title k-Title--noMargin k-Title--italic"
        style={
          Object {
            "--Title-css-color": null,
          }
        }
      >
        Custom title amount
      </h1>
    </div>
    
    <div
      className="k-u-margin-bottom-double k-u-margin-top-double k-u-margin-top-single@xs-down"
    >
      <p
        className="sc-bxivhb iikKd"
      >
        Custom text description
      </p>
    </div>
    
  </div>
</div>
`;

exports[`<CartRewardCard /> ensure backward compatibility matches with snapshot 1`] = `
<div
  className="sc-ifAKCX bpiQjo k-CartRewardCard"
  onAnimationEnd={[Function]}
>
  <div
    className="k-CartRewardCard__cart"
  >
    <button
      aria-label="Delete"
      className="k-ButtonIcon k-CartRewardCard__garbage k-ButtonIcon--hydrogen k-ButtonIcon--micro"
      onClick={[Function]}
      tabIndex={null}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="k-ButtonIcon__svg"
        fill="#fff"
        height="14"
        viewBox="0 0 12 14"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <path
          d="M4.3 0h3v2h-3zM0 2h12v1H0zm1 2l1 10h8l1-10H1zm3.5 8h-1V6h1v6zm2 0h-1V6h1v6zm2 0h-1V6h1v6z"
        />
      </svg>
    </button>
  </div>
  <div
    className="k-u-margin-bottom-quadruple k-CartRewardCard__description"
  >
    <div
      className="k-u-margin-bottom-double k-u-margin-bottom-single@xs-down"
    >
      <h1
        className="sc-bwzfXH AMZiG k-Title k-Title--noMargin k-Title--italic"
        style={
          Object {
            "--Title-css-color": null,
          }
        }
      >
        Custom title amount
      </h1>
    </div>
    
    <div
      className="k-u-margin-bottom-double k-u-margin-top-double k-u-margin-top-single@xs-down"
    >
      <p
        className="sc-bxivhb iikKd"
      >
        Custom text description
      </p>
    </div>
    
  </div>
</div>
`;

exports[`<CartRewardCard /> with some props matches with snapshot 1`] = `
<div
  className="sc-ifAKCX bpiQjo k-CartRewardCard"
  onAnimationEnd={[Function]}
>
  <div
    className="k-CartRewardCard__cart"
  >
    <button
      aria-label="Delete"
      className="k-ButtonIcon k-CartRewardCard__garbage k-ButtonIcon--hydrogen k-ButtonIcon--micro"
      onClick={[Function]}
      tabIndex={null}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="k-ButtonIcon__svg"
        fill="#fff"
        height="14"
        viewBox="0 0 12 14"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <path
          d="M4.3 0h3v2h-3zM0 2h12v1H0zm1 2l1 10h8l1-10H1zm3.5 8h-1V6h1v6zm2 0h-1V6h1v6zm2 0h-1V6h1v6z"
        />
      </svg>
    </button>
  </div>
  <div
    className="k-u-margin-bottom-quadruple k-CartRewardCard__description"
  >
    <div
      className="k-u-margin-bottom-double"
    >
      <Custom title tag
        className="sc-bwzfXH AMZiG k-Title k-Title--noMargin k-Title--italic"
        modifier="quinary"
        style={
          Object {
            "--Title-css-color": null,
          }
        }
      >
        Custom title amount
      </Custom title tag>
    </div>
    <div
      className="k-u-margin-bottom-single k-u-margin-top-double"
    >
      <span
        className="sc-htpNat KtFsv k-u-size-default k-u-weight-bold"
      >
        Custom subtitle
      </span>
    </div>
    <div
      className="k-u-margin-bottom-double k-u-margin-top-double"
    >
      <p
        className="sc-bxivhb iikKd"
      >
        Custom text description
      </p>
    </div>
    <p
      className="sc-htpNat KtFsv k-u-size-tiny k-u-weight-regular k-u-margin-none"
    >
      Custom shipping title
       
      <span
        className="sc-htpNat KtFsv k-u-weight-light"
      >
        Custom shipping value
      </span>
    </p>
    <div
      className="sc-bdVaJa jtfdLD"
    >
      <a
        className="sc-htpNat KtFsv k-u-color-primary1 k-u-decoration-none k-u-size-tiny k-u-weight-regular"
        href="Custom amount link"
      >
        Custom amount
      </a>
    </div>
  </div>
</div>
`;
